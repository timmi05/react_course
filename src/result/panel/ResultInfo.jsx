import React from 'react';

export const ResultInfo = (props) => {

    const { count, genre } = props;
    const massage = count ? `${ count } movie found` : genre ? `Film by ${ genre } genre` : '';

    return (
        <span className='result-info'>
        { massage }
        </span>
    );
}

export default ResultInfo