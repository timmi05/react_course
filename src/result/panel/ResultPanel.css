.result-panel{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 25px 30px 25px;
    font-size: 20px;
    margin: 0 40px;
}
