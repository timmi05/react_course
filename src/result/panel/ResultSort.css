.result-sort{
    display: flex;
}

.result-sort-button{
    font-size: 15px;
    padding: 10px 20px;
}

.result-sort-label{
    padding-right: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}