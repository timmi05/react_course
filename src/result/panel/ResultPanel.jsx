import React from 'react';
import { ResultInfo } from './ResultInfo';
import { ResultSort } from './ResultSort';
import './ResultPanel.css'

export const ResultPanel = props => {
    const { productSelected, genre, data } = props;
    return(
     <div className='result-panel'>
         <ResultInfo
         productSelected = {productSelected}
         genre = { genre }
         data = { data }
         />
         < ResultSort
         productSelected = { productSelected }
         />
    </div>
    )
};
