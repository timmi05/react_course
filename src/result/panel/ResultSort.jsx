import React, { useState } from 'react';
import { Button } from '../../component/Button'
import './ResultSort.css'

export const ResultSort = props => {

    const { productSelected } = props;
    const [disabled, setDisabled] = useState(false);

    const SortByReleaseDate = () => {
        setDisabled(true);
    };

    const SortByRating = () => {
        setDisabled(false);
    };

    if ( productSelected )
    return(
        <a/>
    );

    return (
        <span className='result-sort'>
        <label className='result-sort-label'>SORT BY </label>
            <div className='result-sort-buttons'>
                <Button
                label = 'RELEASE DATE'
                classes = 'result-sort-button result-sort-left-button'
                onClick = { SortByReleaseDate }
                disabled = { disabled }
                />
                <Button
                label = 'RATING'
                classes = 'result-sort-button result-sort-right-button'
                onClick = { SortByRating }
                disabled = { !disabled }
                />
            </div>
        </span>
      );
};
