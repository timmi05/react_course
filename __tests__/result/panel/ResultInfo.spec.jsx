import React from 'react';
import {shallow} from 'enzyme';
import {ResultInfo} from '../../../src/result/panel/ResultInfo';

describe('ResultInfo rendering', () => {
    it('renders if productSelected is true', () => {
        const component = shallow(<ResultInfo
            data={[]}
            isProductSelected={true}
        />);
        expect(component).toMatchSnapshot();
    });

    it('renders if productSelected is false, data empty', () => {
        const component = shallow(<ResultInfo
            data={[]}
            isProductSelected={false}
        />);
        expect(component).toMatchSnapshot();
    });

    it('renders if productSelected is false, data empty, genre not empty', () => {
        const component = shallow(<ResultInfo
            length={0}
            isProductSelected={false}
            genre={'genre'}
        />);
        expect(component).toMatchSnapshot();
    });

    it('renders if productSelected is false, data not empty', () => {
        const component = shallow(<ResultInfo
            length={1}
            isProductSelected={false}
        />);
        expect(component).toMatchSnapshot();
    });
});
