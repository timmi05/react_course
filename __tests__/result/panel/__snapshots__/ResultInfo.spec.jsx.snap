// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultInfo rendering renders if productSelected is false, data empty 1`] = `
<span
  className="result-info"
/>
`;

exports[`ResultInfo rendering renders if productSelected is false, data empty, genre not empty 1`] = `
<span
  className="result-info"
>
  Film by genre genre
</span>
`;

exports[`ResultInfo rendering renders if productSelected is false, data not empty 1`] = `
<span
  className="result-info"
>
  1 movie found
</span>
`;

exports[`ResultInfo rendering renders if productSelected is true 1`] = `
<span
  className="result-info"
/>
`;
